guava/src/com/google/common/hash/Murmur3_32HashFunction.java

```diff
@@ -391,20 +391,23 @@ public HashCode hash() {
   }
 
   private static long codePointToFourUtf8Bytes(int codePoint) {
-    return (((0xFL << 4) | (codePoint >>> 18)) & 0xFF)
+    assert  ((1 << 16) <= codePoint) && (codePoint < (1 << 21)) : "Unexpected codePoint " + codePoint;
+    return ((0xFL << 4) | (codePoint >>> 18))
         | ((0x80L | (0x3F & (codePoint >>> 12))) << 8)
         | ((0x80L | (0x3F & (codePoint >>> 6))) << 16)
         | ((0x80L | (0x3F & codePoint)) << 24);
   }
 
   private static long charToThreeUtf8Bytes(char c) {
-    return (((0xF << 5) | (c >>> 12)) & 0xFF)
+    assert (c < Character.MIN_SURROGATE) || (c > Character.MAX_SURROGATE) : "Unexpected char " + c;
```

Comment:
No particular need for a comment here. In the other cases it is useful for readers to understand that the value couldn't overflow into the mandatory 0 bit after the string of high 1 bits, but here a `char` has 16 bits so after shifting right by 12 that clearly does fit after the high `1110`.

Reply:
From bit shifting perspective -- i agree.
as i were adding input verification, i considered it could be useful here too (for reasons other that bit shifting)

since you recommended comments (vs assert), i removed this